Fix overload names

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #28468 Delete ATenDispatch
* #28186 Remove c10->ATen registration forwarding
* #27978 Migrate remaining ops to the c10 dispatcher
* #28208 Move complex extension test to c10
* #28184 Fix codegen for out operators
* #28183 Allow migrating factory methods to c10
* **#28182 Fix overload names**
* #28181 Add unsupported types to schema type parser

They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

smessmer committed:
Fix overload names

They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

[ghstack-poisoned]

smessmer committed:
Update on "Fix overload names"


They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

[ghstack-poisoned]

ezyang reviewed (approved):
But why wasn't this caught in our existing tests?

smessmer committed:
Update on "Fix overload names"


They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

[ghstack-poisoned]

smessmer commented:
@ezyang because globalATenDispatch doesn't enforce overload names to be unique but c10 does

ezyang commented:
I am suggesting that the native functions parser should check overload names for uniqueness

smessmer committed:
Update on "Fix overload names"


They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

[ghstack-poisoned]

smessmer committed:
Update on "Fix overload names"


They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

[ghstack-poisoned]

smessmer committed:
Update on "Fix overload names"


They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

[ghstack-poisoned]

smessmer committed:
Update on "Fix overload names"


They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

[ghstack-poisoned]

smessmer committed:
Update on "Fix overload names"


They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

[ghstack-poisoned]

smessmer committed:
Update on "Fix overload names"


They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

[ghstack-poisoned]

smessmer committed:
Update on "Fix overload names"


They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

[ghstack-poisoned]

smessmer committed:
Update on "Fix overload names"


They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

[ghstack-poisoned]

smessmer committed:
Update on "Fix overload names"


They haven't been unique. Fixing it...

Differential Revision: [D17969010](https://our.internmc.facebook.com/intern/diff/D17969010/)

[ghstack-poisoned]
